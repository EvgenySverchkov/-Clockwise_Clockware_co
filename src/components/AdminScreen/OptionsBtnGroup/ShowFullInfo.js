import React from 'react';
import {connect} from "react-redux";
import {chngCurrItemForModal} from "../../../store/adminPanel/actions";

function ShowFullInfo({itemObj, chngCurrItemForModal}){
  return (
    <button onClick={()=>{chngCurrItemForModal(itemObj)}} type="button" className="dropdown-item" data-toggle="modal" data-target="#fullInfoModal">
      Show full info
    </button>
  );
}

export default connect(null, {chngCurrItemForModal})(ShowFullInfo);
