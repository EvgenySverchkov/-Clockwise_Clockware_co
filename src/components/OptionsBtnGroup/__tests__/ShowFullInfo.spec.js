import React from "react";
import {create} from "react-test-renderer";
import {Provider} from "react-redux";
import configMockStore from 'redux-mock-store';

import ShowFullInfo from "../ShowFullInfo";

const mockStore = configMockStore();

it("Test of <ShowFullInfo />", ()=>{
    const store = mockStore({})
    const mockProps = {
        itemObj: {
            someField: "Some Value"
        }
    }
    const component = create(
        <Provider store = {store}>
            <ShowFullInfo {...mockProps}/>
        </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
});