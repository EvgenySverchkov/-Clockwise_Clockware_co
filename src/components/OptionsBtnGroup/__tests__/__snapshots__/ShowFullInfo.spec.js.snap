// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <ShowFullInfo /> 1`] = `
<button
  className="dropdown-item"
  data-target="#fullInfoModal"
  data-toggle="modal"
  onClick={[Function]}
  type="button"
>
  Show full info
</button>
`;
